server/src/main/java/org/elasticsearch/cluster/metadata/MetaData.java

```diff
@@ -275,14 +275,12 @@ public boolean equalsAliases(MetaData other) {
 
             if (!filteredValues.isEmpty()) {
                 // Make the list order deterministic
-                CollectionUtil.timSort(filteredValues, new Comparator<AliasMetaData>() {
-                    @Override
-                    public int compare(AliasMetaData o1, AliasMetaData o2) {
-                        return o1.alias().compareTo(o2.alias());
-                    }
-                });
+                CollectionUtil.timSort(filteredValues, Comparator.comparing(AliasMetaData::alias));
+                mapBuilder.put(index, Collections.unmodifiableList(filteredValues));
+            } else if (matchAllAliases) {
```

Comment:
I tried to find a test that covers this branch, at least in IndexAliasesIT I couldn't find any. Is there a test anywhere else? If no, maybe it would make sense to add one.

Reply:
Yes, see my other comment.